### PR TITLE
ContributionFlow: enable float amount

### PR DESCRIPTION
**Pull request by flickz**
- Replace `parseInt` with `parseFloat`
- Change input `step` attribute to `0.01`

Ref [#1988](https://github.com/opencollective/opencollective/issues/1988)

### CODE REVIEW

File: src/components/StyledInputAmount.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const StyledInputAmount = ({ currency, min, max, value, onChange, ...props }) =>
       name="amount"
       maxWidth="10em"
       type="number"
-      step="1"
```
Comment (znarf) [inline src/components/StyledInputAmount.js]: Uhm, how could we have it float and keep a step of 1?

Comment (flickz) [reply src/components/StyledInputAmount.js]: Have tried it, the browser prevents that.

Screenshot from 2019-05-08 17-06-17

Comment (znarf) [reply src/components/StyledInputAmount.js]: Great to know. Wondering if it's better without steps then. What do you think?

No one is gonna use the UI part of it. What are the other benefits? Validation?

Comment (flickz) [reply src/components/StyledInputAmount.js]: The major benefit is validation, if we remove `step` attribute, the browsers will set it by default as `1`. I think it is recommended to set `step` to `0.01` when dealing with money.

https://stackoverflow.com/questions/19011861/is-there-a-float-input-type-in-html5

Comment (znarf) [reply src/components/StyledInputAmount.js]: Cool. Do we have a `min` of 1 now? What happens if you enter `0.25`?

Comment (flickz) [reply src/components/StyledInputAmount.js]: The input `min` attribute is set as `minimumAmount / 100` which some cases `minimumAmount = 100`, hence `min = 1` .

Comment (Betree) [reply src/components/StyledInputAmount.js]: Good to note that precision validation is correctly handled too

image

File: src/components/CreateVirtualCardsForm.js [hunk 2]
```diff
@@ -148,7 +148,7 @@ class CreateVirtualCardsForm extends Component {
         this.setState(state => ({ ...state, values: { ...state.values, numberOfVirtualCards: 1 } }));
       }
     } else if (fieldName === 'amount') {
-      const intAmount = parseInt(value);
+      const intAmount = parseFloat(value);
```
Comment (Betree) [inline src/components/CreateVirtualCardsForm.js]: @flickz `intAmount` is not a really accurate variable name to hold a value returned by `parseFloat` :wink: 

@znarf I'm personally ok with that, do you agree on allowing float amounts for gift cards values?

Comment (znarf) [reply src/components/CreateVirtualCardsForm.js]: @Betree I'm ok with it. In the end, might good to have such flexibility.

Comment (znarf) [reply src/components/CreateVirtualCardsForm.js]: @flickz can you rename `intAmount`?

**Comment by flickz** [comment]
What is delaying this?

**Comment by znarf** [comment]
@flickz Nothing it seems.

**Pull request merged**